/************************************************************************************************************
Module:       hwlog_ft_file

Revision:     1.0

Description:  ---

Notes:        ---

History:
Date          Name      Changes
-----------   ----      -------------------------------------------------------------------------------------
05/16/2023    TH       Began Coding    (TH = Toan Huynh)

COPYRIGHT Â© 2023 TOAN HUYNH. ALL RIGHTS RESERVED.

THIS SOURCE IS TOAN HUYNH PROPRIETARY AND CONFIDENTIAL! NO PART OF THIS
SOURCE MAY BE DISCLOSED IN ANY MANNER TO A THIRD PARTY WITHOUT PRIOR WRITTEN
CONSENT OF TOAN HUYNH.
************************************************************************************************************/

//###########################################################################################################
//      #INCLUDES
//###########################################################################################################
#include <stdint.h>
#include <string.h>
#include <stdbool.h>
#include "file_transfer.h"
#include "hw_log.h"
#include "file-transfer.pb.h"




//###########################################################################################################
//      TESTING #DEFINES
//###########################################################################################################



//###########################################################################################################
//      CONSTANT #DEFINES
//###########################################################################################################



//###########################################################################################################
//      MACROS
//###########################################################################################################



//###########################################################################################################
//      MODULE TYPES
//###########################################################################################################



//###########################################################################################################
//      CONSTANTS
//###########################################################################################################



//###########################################################################################################
//      MODULE LEVEL VARIABLES
//###########################################################################################################
static ft_send_info_t hwlog_file_info =
{
  .session_id = 0,
  .type = fileTransferPkg_fileType_HW_LOG,
  .del_when_done = true,
  .open = hwlog_open,
  .close = hwlog_close,
  .delete = hwlog_delete,
  .file_size = hwlog_get_file_size,
  .read = hwlog_read,
  .image_size = NULL
};



//###########################################################################################################
//      PRIVATE FUNCTION PROTOTYPES
//###########################################################################################################



//###########################################################################################################
//      PUBLIC FUNCTIONS
//###########################################################################################################
/********************************************************************************
Input:
  ---
Output:
  ---
Description:
  ---
Author, Date:
  Toan Huynh, 05/16/2023
*********************************************************************************/
bool hwlog_ft_file_init(void)
{
  return file_transfer_register(&hwlog_file_info, fileTransferPkg_fileType_HW_LOG);
}

//###########################################################################################################
//      PRIVATE FUNCTIONS
//###########################################################################################################


//###########################################################################################################
//      INTERRUPTS
//###########################################################################################################



//###########################################################################################################
//      TEST HARNESSES
//###########################################################################################################



//###########################################################################################################
//      END OF hwlog_ft_file.c
//###########################################################################################################
